util/fp_finder.go

```diff
@@ -0,0 +1,209 @@
+	// Get the dictionary path in ~/.crs-toolchain
+	dictionaryPath, err := t.getDictionaryPath()
+	if err != nil {
+		logger.Fatal().Err(err).Msgf("Error getting dictionary path: %v", err)
+	}
+
+	// Check if the dictionary exists, if not, download it
+	if _, err := os.Stat(dictionaryPath); os.IsNotExist(err) {
+		logger.Debug().Msgf("Dictionary file not found. Downloading...")
+		if err := t.downloadFile(dictionaryPath, dictionaryURL); err != nil {
+			logger.Fatal().Err(err).Msgf("Failed to download dictionary: %v", err)
+		}
+		logger.Debug().Msgf("Download complete.")
+	} else {
+		logger.Debug().Msgf("Dictionary file found, skipping download.")
+	}
+
+	// Load dictionary into memory
+	dict, err := t.loadDictionnary(dictionaryPath, minSize)
+	if err != nil {
+		logger.Fatal().Err(err).Msgf("Failed to load dictionary: %v", err)
+	}
+
+	// Load input file into memory
+	inputFile, err := t.loadFileContent(inputFilePath)
+	if err != nil {
+		logger.Fatal().Err(err).Msgf("Failed to load input file: %v", err)
+	}
+
+	// Filter words not in dictionary, remove duplicates, and sort alphabetically
+	filteredWords := t.filterContent(inputFile, dict, minSize)
+
+	if uniqEnabled {
+		filteredWords = t.removeDuplicates(filteredWords)
+	}
+
+	if sortEnabled {
+		sort.Slice(filteredWords, func(i, j int) bool {
+			return strings.ToLower(filteredWords[i]) < strings.ToLower(filteredWords[j])
+		})
+	}
+
+	for _, str := range filteredWords {
+		fmt.Println(str)
+	}
+
+	return nil
+}
+
+func (t *FpFinder) getDictionaryPath() (string, error) {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+	crsToolchainDir := filepath.Join(homeDir, ".crs-toolchain")
+
+	// Create ~/.crs-toolchain folder if it doesn't exist
+	if _, err := os.Stat(crsToolchainDir); os.IsNotExist(err) {
+		if err := os.MkdirAll(crsToolchainDir, 0755); err != nil {
+			return "", err
+		}
+	}
+
+	return filepath.Join(crsToolchainDir, dictionaryFileName), nil
+}
+
+func (t *FpFinder) downloadFile(filepath, url string) error {
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("bad status: %s", resp.Status)
+	}
+
+	_, err = io.Copy(out, resp.Body)
+	return err
+}
+
+func (t *FpFinder) loadDictionnary(path string, minSize int) (map[string]struct{}, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	content := make(map[string]struct{})
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		word := strings.TrimSpace(scanner.Text())
+		if len(word) >= minSize {
+			content[word] = struct{}{}
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	// Add words from the embedded extendedDictionnary
+	for _, word := range strings.Split(extendedDictionnary, "\n") {
```

Comment:
Use a builtin / library function to split lines. It will probably be fine in this case, but some day the file may be committed from a Windows machine...